source/Halibut/Transport/Protocol/WebSocketStream.cs

```diff
@@ -49,13 +50,24 @@ public async Task<string> ReadTextMessage()
             var buffer = new ArraySegment<byte>(new byte[10000]);
             while (true)
             {
-                var result = await context.ReceiveAsync(buffer, cancel.Token);
-                if (result.MessageType != WebSocketMessageType.Text)
-                    throw new Exception($"Encountered an unexpected message type {result.MessageType}");
-                sb.Append(Encoding.UTF8.GetString(buffer.Array, 0, result.Count));
-
-                if (result.EndOfMessage)
-                    return sb.ToString();
+                using(var cts = new CancellationTokenSource(HalibutLimits.TcpClientReceiveTimeout))
```

Comment:
Drops connections that have not sent data after 10 minutes... don't know if that is too long